source/common/network/connection_impl.cc

```diff
@@ -923,6 +924,11 @@ void ClientConnectionImpl::connect() {
 
 void ClientConnectionImpl::onConnected() {
   stream_info_.upstreamInfo()->upstreamTiming().onUpstreamConnectComplete(dispatcher_.timeSource());
+  // There are no meaningful socket source address semantics for non-IP sockets, so skip.
```

Comment:
Or we can make the `set*`  API take a optional arg `absl::optional<absl::string_view>`. Then we only need do simple check in the `set*` API implementation. 🤔

Reply:
> Would interfaceName() just be empty in that case? I wonder if it would be simpler semantically to gate on just that

No, unfortunately this goes all the way to what unix provides us in `getsockname`: a successful syscall but an non-zero bytes "empty" string. So, as it stands calling `localAddress()` on a UDS throws. I can fix that. But it seemed independent of this change.